paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentOptionsStateFactoryTest.kt

```diff
@@ -171,6 +174,78 @@ class PaymentOptionsStateFactoryTest {
         assertThat(options[2].isEnabledDuringEditing).isTrue()
     }
 
+    @Test
+    fun `State is correct when given non null defaultPaymentMethodId and it is selected`() {
```

Comment:
a couple ideas to simplify these tests:
- is it necessary to test the selection state? It doesn't look like being selected or not will influence whether shouldShowDefaultBadge is true, so I suspect we could leave that out
- I think we could probably combine this test and the one after it -- e.g. we could say the defaultPaymentMethodId is equal to the first payment method's ID. And then we can assert that the first payment method's `shouldShowDefaultBadge` is true, and the other one's are false